module-bsp/board/rt1051/common/WorkerQueue.hpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#pragma once
+
+#include "FreeRTOS.h"
+#include "task.h"
+#include "queue.h"
+
+#include <functional>
+#include <utility>
+
+/*
+ * Worker queue implementation is used to defer work from interrupt routines.
+ * This way, it is possible to keep interrupt routines short but at the same time process more intensive tasks.
+ * The design goal was to use as few resources as possible, hence it is a lightweight mechanism.
+ * The stack assigned to the queue worker handler is relatively small.
+ * It is not suitable for performing stack-consuming tasks, for instance handling I/O.
+ * If necessary, stack size can be manually reconfigured upon creation.
+ */
+template <typename Message> class WorkerQueue
```

Comment:
I've dejavu as long time ago we were talking about the fact that all Workers in Services should work with some better WorkerQueue mechanism :) (better than generally used one )

Reply:
Was this architectural approach discussed with the team/system chapter?

Reply:
This solution is nothing new. We are already using it in the existing event manager, just in the uglier way(breaking SRP and encapsulation). The goal here was to create and use bsp-related os primitives(for now I only added the queue implementation) and do not mix services/app-related things with low-level code(try to think about it as the kernel). We discussed it briefly with @pholat and have a plan of what to do next.